kernel/src/main/scala-2.12/cats/kernel/compat/unused.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2015 Typelevel
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package cats.kernel.compat
+
+/** Simulates the behaviour of `@unused` annotation originally implemented
+  * in Scala 2.13 and above.
+  * 
+  * @note copied from
+  *       [[https://github.com/scala/scala-collection-compat/blob/e35e330fca28c78c39fa2d4812fa33ce892f19e3/compat/src/main/scala-2.11_2.12/scala/annotation/unused.scala scala-collection-compat]]
+  *       source code.
+  */
+final private[cats] class unused extends deprecated("unused", "unused")
```

Comment:
Just to draw more attention to this Scala version compatibility trick.

Reply:
I'm on a fence with this implementation of `unused` for Scala 2.12.
Turns out that it does not always work out everywhere (due to 2.12 limitations I guess). For example, in this case (let's name it "A")
https://github.com/typelevel/cats/blob/f8f6eff94872cba8bead3ae967d5b4f60f1b068d/core/src/main/scala/cats/data/Const.scala#L42-L43
it works just fine.

But on the other hand for this case "B":
https://github.com/typelevel/cats/blob/f8f6eff94872cba8bead3ae967d5b4f60f1b068d/core/src/main/scala/cats/ApplicativeError.scala#L356-L362

it produces another quite weird warning (instead of "parameter is never used"):
```
[warn] /Users/.../typelevel/cats/core/src/main/scala/cats/ApplicativeError.scala:359:7: value NT is deprecated
[warn]       try {
[warn]       ^
```
And there are a few more similar cases in `core`. I figured out that for the latter case ("B") the following implementation of `@unused` works better:
```scala
final private[cats] class unused extends scala.annotation.nowarn("msg=is never used")
```
But it does not work at all for the former case ("A") – Scala 2.12 warns the `@nowarn` annotation does not suppress any warning.

So seems that to  handle differences between 2.12 and 2.13 properly we need to have 2 different compatibility implementations for `@unused`: (1) `@deprecated`-based and (2) `@nowarn`-based. And pick one of them depending on a particular case.

@armanbilge any thoughts on this?